test/e2e/service.go

```diff
@@ -2350,3 +2371,33 @@ func execSourceipTest(f *framework.Framework, c *client.Client, ns, nodeName, se
 
 	return execPodIp, sourceIp
 }
+
+func getLoadBalancerName(service *api.Service) string {
+	//GCE requires that the name of a load balancer starts with a lower case letter.
+	ret := "a" + string(service.UID)
+	ret = strings.Replace(ret, "-", "", -1)
+	//AWS requires that the name of a load balancer is shorter than 32 bytes.
+	if len(ret) > 32 {
+		ret = ret[:32]
+	}
+	return ret
+}
+
+func cleanupServiceGCEResources(loadBalancerName string) {
+	if pollErr := wait.Poll(5*time.Second, lbCleanupTimeout, func() (bool, error) {
+		if err := framework.CleanupGCEResources(loadBalancerName); err != nil {
```

Comment:
the way we do this in ingress e2e is first poll on the resources and wait for the servicecontroller to delete them in, say, 5m. If this happens, test pass. Otherwise test fails, but at the end we delete the resources ourselves.